metrics/web/docs/features/router_widget/router_widget.md

```diff
@@ -0,0 +1,147 @@
+# Router widget usage design
+
+> Summary of the proposed change
+
+Use the Router widget instead of Navigator on the Web App.
+
+# References
+
+> Link to supporting documentation, GitHub tickets, etc.
+* [Learning Flutter’s new navigation and routing system](https://medium.com/flutter/learning-flutters-new-navigation-and-routing-system-7c9068155ade)
+
+* [Router Class](https://api.flutter.dev/flutter/widgets/Router-class.html)
+
+* [MaterialApp.router](https://api.flutter.dev/flutter/material/MaterialApp/MaterialApp.router.html)
+
+* [Example](https://github.com/orestesgaolin/navigator_20_example/blob/master/lib/main_router.dart)
+
+# Motivation
+
+> What problem is this project solving?
+
+Make UX more convenient.
+
+# Goals
+
+> Identify success metrics and measurable goals.
+
+Correct behavior of the browser features as a forward button or the browser history.
+
+# Non-Goals
+
+> Identify what's not in scope.
+
+The custom animation transition and the functionality of the existing observers are not in scope.
+
+# Design
+
+> Explain and diagram the technical design
+
+## Metrics application
+
+### Presentation layer
+
```

Comment:
Let's divide this section into the following sub-sections: 
- #### Making the app to use the Router class
Provide instructions on how to configure the application to use the Router. Do we need to use `MaterialApp.router`? What to do with observers then? How to replace or move parameters that are not presented in `MaterialApp.router`? Maybe this sub-section should be last. 
- #### Defining app routes
Provide instructions and design on how we can define app routes. Would it be a class with routes' names? How to add a new route? How to define our existing route? Think about how to make adding a new route easy and SOLID.
- #### Parsing a route information
Here we'll state how to parse route information (creating MetricsRouteInformationParser).
- #### Creating a route delegate and routes state
Describe here how to create a delegate and how to change the state of the navigator. Not sure that combining state and `build` in the same class is a good idea. It seems like we can have a separate notifier for that logic as described in the [example](https://github.com/orestesgaolin/navigator_20_example/blob/master/lib/main_router.dart). 
- #### Making things work
This sub-section states the full design, the algorithm of routing (sequence diagram would be a great plus). 

No need for big sections with code - instead, provide short code samples with parts that are not as clear as expected. The main idea is to make the feature clear and understandable by everyone in the team without googling it.